File: checkio_js/elementary/javascript/correctSentence.js
import assert from "assert";

function correctSentence(text) {
    if (text.match(/^[a-z]/) != null){
        text = text[0].toUpperCase() + text.substring(1)
    }
    if (text.match(/[.]$/) == null){
        text += '.'
    }
    return text;
}

console.log('Example:');
console.log(correctSentence('greetings, friends'));

// These "asserts" are used for self-checking
assert.equal(correctSentence('greetings, friends'), 'Greetings, friends.');
assert.equal(correctSentence('Greetings, friends'), 'Greetings, friends.');
assert.equal(correctSentence('Greetings, friends.'), 'Greetings, friends.');

console.log("Coding complete? Click 'Check' to earn cool rewards!");

File: checkio_js/elementary/python/correctSentence.py


def correctSentence(text):
    if text[-1] != '.':
        return f'{text[0].upper()}{text[1:]}.'
    return f'{text[0].upper()}{text[1:]}'

if __name__ == '__main__':
    print('Example:')
    print(correctSentence('greetings, friends'))

# These "asserts" are used for self-checking
assert correctSentence('greetings, friends') == 'Greetings, friends.'
assert correctSentence('Greetings, friends') == 'Greetings, friends.'
assert correctSentence('Greetings, friends.') == 'Greetings, friends.'

print("Coding complete? Click 'Check' to earn cool rewards!")
